core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/TransmissionNetworkOutput.java

```diff
@@ -176,6 +176,9 @@ public boolean send(Transmission transmission) {
                 if (ioe instanceof ConnectTimeoutException) {
                     transmissionPolicyManager.suspendInSeconds(TransmissionPolicy.BLOCKED_BUT_CAN_BE_PERSISTED, DEFAULT_BACKOFF_TIME_SECONDS);
                 }
+            } catch (IllegalStateException e) {
+                suspendTransmissions(TransmissionPolicy.BLOCKED_AND_CANNOT_BE_PERSISTED, response);
```

Comment:
This is correct, and my changes will not work. I fixed the "forever loop bug" in 1.0.8 by adding transmissionPolicyManager.suspendInSeconds(...) I did not notice that suspendTransmissions() is very different from transmissionPolicyManager.suspendInSeconds(..) that is master branch, but I have tested and the forever loop also occure with latest code (10.07.2017)